data/block/block.go

```diff
@@ -581,3 +582,39 @@ func (h *Header) GetAdditionalData() headerVersionData.HeaderAdditionalData {
 	// no extra data for the initial version of shard block header
 	return nil
 }
+
+// CheckFieldsForNil checks a predefined set of fields for nil values
+func (h *Header) CheckFieldsForNil() error {
+	if h == nil {
+		return data.ErrNilPointerReceiver
+	}
+	if h.PrevHash == nil {
+		return fmt.Errorf("%w in Header.PrevHash", data.ErrNilValue)
+	}
+	if h.PrevRandSeed == nil {
+		return fmt.Errorf("%w in Header.PrevRandSeed", data.ErrNilValue)
+	}
+	if h.RandSeed == nil {
+		return fmt.Errorf("%w in Header.RandSeed", data.ErrNilValue)
+	}
+	if h.PubKeysBitmap == nil {
+		return fmt.Errorf("%w in Header.PubKeysBitmap", data.ErrNilValue)
+	}
+	if h.RootHash == nil {
+		return fmt.Errorf("%w in Header.RootHash", data.ErrNilValue)
+	}
+	if h.ChainID == nil {
+		return fmt.Errorf("%w in Header.ChainID", data.ErrNilValue)
+	}
+	if h.SoftwareVersion == nil {
+		return fmt.Errorf("%w in Header.SoftwareVersion", data.ErrNilValue)
+	}
+	if h.AccumulatedFees == nil {
+		return fmt.Errorf("%w in Header.AccumulatedFees", data.ErrNilValue)
+	}
+	if h.DeveloperFees == nil {
+		return fmt.Errorf("%w in Header.DeveloperFees", data.ErrNilValue)
+	}
```

Comment:
Signature and LeaderSignature should not be checked for nil? both here and in the metablock?

Reply:
I do not think so. Reason: the proposed block that is sent on the consensus topic does not set those 2 fields

Reply:
ok

Reply:
Also PubKeyBitmap is nil on the proposed header.

Reply:
+1